service/tokenizer.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2021 The casbin Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"sync"
+	"unicode/utf8"
+
+	"github.com/huichen/sego"
+)
+
+var segmenter sego.Segmenter
+var onceload sync.Once
+
+func Finalword(s string) []string {
+	Loaddic()
+	return Keyword(Tokenizer(s))
+}
+
+func Loaddic() {
+	onceload.Do(func() {
+		segmenter.LoadDictionary("./dictionary/dictionary.txt")
+	})
+}
+
+func Tokenizer(s string) []string {
+	text := []byte(s)
+	segments := segmenter.Segment(text)
+	return sego.SegmentsToSlice(segments, false)
+}
+
+func Keyword(word []string) []string {
+	var filterword, sortword, finalword []string
+	var sortnumber []int
+	wordnum := make(map[string]int)
+	for _, val := range word {
+		if utf8.RuneCountInString(val) > 1 {
+			filterword = append(filterword, val)
+		}
+	}
+	if len(filterword) == 0 {
+		return word
```

Comment:
If `filterword == 0` (means no filterword length logger than 1), but `len(word)` could be more than 5.